From 8a472a23a53d00752fc12000dcac2acaa0b6908b Mon Sep 17 00:00:00 2001
From: David Mitchell <davem@iabyn.com>
Date: Wed, 25 Mar 2015 16:04:58 +0000
Subject: perlfunc: mention that study() is currently a noop

(cherry picked from commit d8c1af4e0d4774b60bd59dc0039c7e101f292a4a)

Bug-Debian: https://bugs.debian.org/822336
Patch-Name: fixes/5.20.3/docs/study_noop.diff
---
 pod/perlfunc.pod | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/pod/perlfunc.pod b/pod/perlfunc.pod
index 802b99080..272a6cd0a 100644
--- a/pod/perlfunc.pod
+++ b/pod/perlfunc.pod
@@ -7671,14 +7671,18 @@ X<study>
 
 =for Pod::Functions optimize input data for repeated searches
 
-Takes extra time to study SCALAR (C<$_> if unspecified) in anticipation of
-doing many pattern matches on the string before it is next modified.
+May take extra time to study SCALAR (C<$_> if unspecified) in anticipation
+of doing many pattern matches on the string before it is next modified.
 This may or may not save time, depending on the nature and number of
 patterns you are searching and the distribution of character
 frequencies in the string to be searched; you probably want to compare
 run times with and without it to see which is faster.  Those loops
 that scan for many short constant strings (including the constant
 parts of more complex patterns) will benefit most.
+
+Note that since Perl version 5.16 this function has been a no-op, but
+this might change in a future release.
+
 (The way C<study> works is this: a linked list of every
 character in the string to be searched is made, so we know, for
 example, where all the C<'k'> characters are.  From each search string,
